### PR TITLE
Default2

### PR DESCRIPTION
**Pull request by a-siva**

### CODE REVIEW

File: packages/flutter/test/foundation/error_reporting_test.dart [hunk 1]
```diff
@@ -45,7 +43,7 @@ Future<Null> main() async {
 
   final StackTrace sampleStack = await getSampleStack();
 
-  final String dividerRegExp = pathSeparatorForRegExp;
+  const String dividerRegExp = '/';
```
Comment (Hixie) [inline packages/flutter/test/foundation/error_reporting_test.dart]: really?

File: packages/flutter_tools/lib/src/commands/build_apk.dart [hunk 2]
```diff
@@ -15,7 +15,10 @@ class BuildApkCommand extends BuildSubCommand {
     usesPubOption();
 
     argParser
-      ..addFlag('preview-dart-2', negatable: false,  hide: !verboseHelp)
+      ..addFlag('preview-dart-2',
+          defaultsTo: true,
+          negatable: true,
+	  hide: !verboseHelp)
```
Comment (Hixie) [inline packages/flutter_tools/lib/src/commands/build_apk.dart]: indentation style here doesn't match build_aot.dart's style; we should be consistent. :-)

File: packages/flutter_tools/lib/src/commands/build_flx.dart [hunk 3]
```diff
@@ -21,7 +21,11 @@ class BuildFlxCommand extends BuildSubCommand {
     argParser.addOption('snapshot', defaultsTo: defaultSnapshotPath);
     argParser.addOption('depfile', defaultsTo: defaultDepfilePath);
     argParser.addOption('kernel-file', defaultsTo: defaultApplicationKernelPath);
-    argParser.addFlag('preview-dart-2', negatable: false, hide: !verboseHelp);
+    argParser.addFlag(
+      'preview-dart-2',
+      defaultsTo: true,
+      negatable: true,
+      hide: !verboseHelp);
```
Comment (Hixie) [inline packages/flutter_tools/lib/src/commands/build_flx.dart]: ditto

File: packages/flutter_tools/lib/src/commands/build_ios.dart [hunk 4]
```diff
@@ -29,7 +29,7 @@ class BuildIOSCommand extends BuildSubCommand {
     argParser.addFlag('simulator', help: 'Build for the iOS simulator instead of the device.');
     argParser.addFlag('codesign', negatable: true, defaultsTo: true,
         help: 'Codesign the application bundle (only available on device builds).');
-    argParser.addFlag('preview-dart-2', negatable: false,
+    argParser.addFlag('preview-dart-2', defaultsTo: true, negatable: true,
```
Comment (Hixie) [inline packages/flutter_tools/lib/src/commands/build_ios.dart]: ditto

File: packages/flutter_tools/lib/src/commands/drive.dart [hunk 5]
```diff
@@ -75,7 +75,8 @@ class DriveCommand extends RunCommandBase {
     );
 
     argParser.addFlag('preview-dart-2',
-        defaultsTo: false,
+        defaultsTo: true,
+        negatable: true,
```
Comment (Hixie) [inline packages/flutter_tools/lib/src/commands/drive.dart]: there are yet a different style, d'oh

File: packages/flutter_tools/lib/src/commands/build_aot.dart [hunk 6]
```diff
@@ -39,7 +39,11 @@ class BuildAotCommand extends BuildSubCommand {
       )
       ..addFlag('interpreter')
       ..addFlag('quiet', defaultsTo: false)
-      ..addFlag('preview-dart-2', negatable: false, hide: !verboseHelp)
+      ..addFlag('preview-dart-2',
+        defaultsTo: true,
+        negatable: true,
```
Comment (Hixie) [inline packages/flutter_tools/lib/src/commands/build_aot.dart]: here and many cases below: i'm pretty sure "negatable: true" is the default

File: packages/flutter_tools/lib/src/compile.dart [hunk 7]
```diff
@@ -12,8 +12,12 @@ import 'base/io.dart';
 import 'base/process_manager.dart';
 import 'globals.dart';
 
+typedef void CompilerMessageConsumer(String message);
+
 class _StdoutHandler {
-  _StdoutHandler() {
+  final CompilerMessageConsumer consumer;
+
+  _StdoutHandler({this.consumer: printError}) {
```
Comment (Hixie) [inline packages/flutter_tools/lib/src/compile.dart]: constructors first

File: packages/flutter_tools/lib/src/test/flutter_platform.dart [hunk 8]
```diff
@@ -108,6 +108,29 @@ class _Compiler {
...
+
+      printError('$message');
+    }
+
+    ResidentCompiler createCompiler() {
+      return new ResidentCompiler(
+          artifacts.getArtifactPath(Artifact.flutterPatchedSdkPath),
+          packagesPath: PackageMap.globalPackagesPath,
+          trackWidgetCreation: trackWidgetCreation,
+          compilerMessageConsumer: reportCompilerMessage);
```
Comment (Hixie) [inline packages/flutter_tools/lib/src/test/flutter_platform.dart]: please use flutter style for indentation of arguments (matches dartfmt's style with a trailing comma)

File: packages/flutter_tools/lib/src/test/flutter_platform.dart [hunk 9]
```diff
@@ -118,28 +141,34 @@ class _Compiler {
...
+          // Check if the compiler produced the output. It could have crashed
+          // in which case outputPath would be null. In this case pass
+          // null upwards to the consumer and shutdown the compiler.
+          if (outputPath == null) {
+            request.result.complete(null);
+            compiler.shutdown();
+            compiler = null;
+          } else {
+            final File kernelReadyToRun = await fs.file(outputPath).copy(
               request.path + '.dill');
```
Comment (Hixie) [inline packages/flutter_tools/lib/src/test/flutter_platform.dart]: ditto

File: packages/flutter_tools/test/commands/create_test.dart [hunk 10]
```diff
@@ -477,11 +477,16 @@ Future<Null> _runFlutterTest(Directory workingDir, {String target}) async {
...
+  print('$dartSdkPath/bin/dart');
+  print(args);
+  print(workingDir.path);
+  final ProcessResult exec = Process.runSync(
     '$dartSdkPath/bin/dart',
     args,
     workingDirectory: workingDir.path,
   );
+  print(exec.stdout);
+  print(exec.stderr);
```
Comment (Hixie) [inline packages/flutter_tools/test/commands/create_test.dart]: i'm assuming these prints are accidental?

File: packages/flutter_tools/templates/plugin/lib/projectName.dart.tmpl [hunk 11]
```diff
@@ -6,6 +6,8 @@ class {{pluginDartClass}} {
   static const MethodChannel _channel =
       const MethodChannel('{{projectName}}');
 
-  static Future<String> get platformVersion =>
-      _channel.invokeMethod('getPlatformVersion');
+  static Future<String> get platformVersion async {
+    final String version = await _channel.invokeMethod('getPlatformVersion');
+    return version;
```
Comment (Hixie) [inline packages/flutter_tools/templates/plugin/lib/projectName.dart.tmpl]: this looks like the remains of debugging? (as far as i can tell the new code is just a more verbose version of the old code)

File: dev/bots/test.dart [hunk 12]
```diff
@@ -130,15 +129,6 @@ Future<Null> _analyzeRepo() async {
   print('${bold}DONE: Analysis successful.$reset');
 }
 
-Future<Null> _runTestsDart2() async {
```
Comment (tvolkert) [inline dev/bots/test.dart]: We should probably run a shard with `--no-preview-dart-2` now.

Comment (tvolkert) [reply dev/bots/test.dart]: Although I guess that would preclude us from big cleanups, like new/const removal... nevermind 😄

**Comment by Hixie** [comment]
As it stands this will regress the `debug_full_compile_millis` benchmark by ~10% and the `hotReloadDevFSSyncMillisecondsAfterChange` benchmark (i.e. hot reload) by about 100ms in the median case. We shouldn't land this until we are at least on par with current performance metrics.

File: packages/flutter_tools/templates/plugin/lib/projectName.dart.tmpl (hunk 11 above)
Comment (mraleph) [reply packages/flutter_tools/templates/plugin/lib/projectName.dart.tmpl]: `invokeMethod` returns `Future`, not `Future<String>` so in Dart 2 the old form does not work.

File: dev/bots/test.dart (hunk 12 above)
Comment (mraleph) [reply dev/bots/test.dart]: We want to have that temporarily, but when I tried it I discovered that it does not fully work. Some of the stuff that `_runTests` invokes did not understand `--no-preview-dart-2`. In any case we most likely are not gonna support `--no-preview-dart-2` for long. If Flutter framework starts using features like reified generic methods or optional new/const, there is no way to run this on Dart 1.

**Review by mraleph — COMMENTED** [review]
@Hixie I have addressed all your comments as part of the original PR https://github.com/flutter/flutter/pull/15565

This PR was just WIP from @a-siva trying to debug AppVeyor issue. (Which should be fixed on my PR as well now). 

Feel free to continue review there.

**Pull request closed without merging**